src/librustc_typeck/check/op.rs

```diff
@@ -191,6 +191,27 @@ fn check_overloaded_binop<'a, 'tcx>(fcx: &FnCtxt<'a, 'tcx>,
                               "binary operation `{}` cannot be applied to type `{}`",
                               hir_util::binop_to_string(op.node),
                               lhs_ty);
+                    let (display_note, missing_trait) = match op.node {
+                        hir::BiAdd    => (true, "std::ops::Add"),
```

Comment:
This may be more idiomatically done via `Some("...")` with a `None` at the end followed by a `if let Some(missing_trait) = display_note` at the printing location.